xen/drivers/passthrough/iommu.c

```diff
@@ -249,22 +249,39 @@ void iommu_domain_destroy(struct domain *d)
     arch_iommu_domain_destroy(d);
 }
 
-int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
-                   unsigned int flags)
+int iommu_map_pages(struct domain *d, unsigned long gfn, unsigned long mfn,
+                    unsigned long page_count, unsigned int flags)
 {
     const struct domain_iommu *hd = dom_iommu(d);
-    int rc;
+    int rc = 0;
```

Comment:
Initializing rc with -ENOSYS is perfectly fitting variant here: it will be remained only in a case someone missed defining map_page/map_pages. This completely corresponds to "we should not be here" case.